autonomy/chain/mint.py

```diff
@@ -275,7 +280,10 @@ def mint_service(  # pylint: disable=too-many-arguments
             raise InvalidMintParameter("Cost of bond cannot be zero")
 
         number_of_agent_instances = sum(number_of_slots_per_agent)
-        if threshold < (ceil((number_of_agent_instances * 2 + 1) / 3)):
+        if threshold is None:
+            threshold = get_min_threshold(number_of_agent_instances)
+
+        if threshold < get_min_threshold(n=number_of_agent_instances):
```

Comment:
Threshold will be calculated automatically using the number of agents if the user does not provide a value